packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -1514,6 +1514,15 @@ export default class RestApi {
         res.sendStatus(200)
       })
     )
+
+    api.delete(
+      '/:collection(users)/:id',
+      wrap(async (req, res) => {
+        const { id } = req.params
+        await app.deleteUser(id)
```

Comment:
In case the user do not exist it will throw and lead to an `internal server error (500)`.
Catch and return `not found (404)` in that case